app/src/main/java/org/mozilla/fenix/home/sessioncontrol/viewholders/onboarding/MessageCardViewHolder.kt

```diff
@@ -5,53 +5,49 @@
 package org.mozilla.fenix.home.sessioncontrol.viewholders.onboarding
 
 import android.view.View
-import androidx.core.view.isVisible
-import androidx.recyclerview.widget.RecyclerView
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.platform.ComposeView
+import androidx.lifecycle.LifecycleOwner
+import mozilla.components.lib.state.ext.observeAsComposableState
 import org.mozilla.fenix.R
-import org.mozilla.fenix.databinding.NimbusMessageCardBinding
-import org.mozilla.fenix.ext.increaseTapArea
-import org.mozilla.fenix.gleanplumb.Message
+import org.mozilla.fenix.components.components
+import org.mozilla.fenix.compose.ComposeViewHolder
+import org.mozilla.fenix.compose.MessageCard
 import org.mozilla.fenix.home.sessioncontrol.SessionControlInteractor
 
+/**
+ * View holder for the Nimbus Message Card.
+ *
+ * @property interactor [SessionControlInteractor] which will have delegated to all user
+ * interactions.
+ */
 class MessageCardViewHolder(
-    view: View,
-    private val interactor: SessionControlInteractor
-) : RecyclerView.ViewHolder(view) {
+    composeView: ComposeView,
+    viewLifecycleOwner: LifecycleOwner,
+    private val interactor: SessionControlInteractor,
+) : ComposeViewHolder(composeView, viewLifecycleOwner) {
 
-    fun bind(message: Message) {
-        val binding = NimbusMessageCardBinding.bind(itemView)
-
-        if (message.data.title.isNullOrBlank()) {
-            binding.titleText.isVisible = false
-        } else {
-            binding.titleText.text = message.data.title
-        }
-
-        binding.descriptionText.text = message.data.text
-
-        if (message.data.buttonLabel.isNullOrBlank()) {
-            binding.messageButton.isVisible = false
-            binding.experimentCard.setOnClickListener {
-                interactor.onMessageClicked(message)
-            }
-        } else {
-            binding.messageButton.text = message.data.buttonLabel
-            binding.messageButton.setOnClickListener {
-                interactor.onMessageClicked(message)
-            }
-        }
+    companion object {
+        internal val LAYOUT_ID = View.generateViewId()
+    }
 
-        binding.close.apply {
-            increaseTapArea(CLOSE_BUTTON_EXTRA_DPS)
-            setOnClickListener {
-                interactor.onMessageClosedClicked(message)
-            }
-        }
-        interactor.onMessageDisplayed(message)
```

Comment:
Looks like in the new implementation we are missing this call.

Reply:
Thanks @Amejia481 , good catch! Missed it when removed the bind implementation .